src/NServiceBus.Core/Transports/Msmq/MsmqMessageDispatcher.cs

```diff
@@ -129,12 +129,27 @@ bool IsCombiningTimeToBeReceivedWithTransactions(ReadOnlyContextBag context, Dis
                 return false;
             }
 
+            if (Transaction.Current != null)
```

Comment:
Fair point

On Friday, February 12, 2016, Andreas Öhlund <notifications@github.com>
wrote:

> In src/NServiceBus.Core/Transports/Msmq/MsmqMessageDispatcher.cs
> <https://github.com/Particular/NServiceBus/pull/3460#discussion_r52771054>
> :
>
> > @@ -129,12 +129,27 @@ bool IsCombiningTimeToBeReceivedWithTransactions(ReadOnlyContextBag context, Dis
> >                  return false;
> >              }
> >
> > +            if (Transaction.Current != null)
>
> The msmq transport already adds it as well.
>
> Regarding the check we need to check for a ambient scope since the user
> might wrap his sends in a scope when executing outside of a handler.
>
> This pull didn't change this in anyway, lets discuss this in a separate
> issue?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3460/files#r52771054>.
>